sdk/inc/azure/core/az_json.h

```diff
@@ -359,6 +359,34 @@ az_json_writer_get_bytes_used_in_destination(az_json_writer const* json_writer)
  */
 AZ_NODISCARD az_result az_json_writer_append_string(az_json_writer* json_writer, az_span value);
 
+/**
+ * @brief Appends an existing UTF-8 encoded JSON text into the buffer, useful for appending nested
+ * JSON.
+ *
+ * @param[in] json_writer A pointer to an #az_json_writer instance containing the buffer to append
+ * the JSON text to.
+ * @param[in] json_text A single, possibly nested, valid, UTF-8 encoded, JSON value to be written as
+ * is. No modifications are made to this text, including escaping.
+ *
+ * @remarks A single, possibly nested, JSON value is one that starts and ends with {} or [] or is a
+ * single primitive token. The JSON cannot start with an end object or array, or a property name, or
+ * be incomplete.
+ *
+ * @remarks The caller must make sure that the JSON to be appended provided by \p json_text is valid
+ * and properly escaped.
+ *
+ * @return An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if the JSON text value was appended successfully
+ *         - #AZ_ERROR_INSUFFICIENT_SPAN_SIZE if the buffer is too small
+ *         - #AZ_ERROR_JSON_INVALID_STATE if the \p json_writer is in a state where the \p json_text
+ * cannot be appended because it would result in invalid JSON.
+ *         - #AZ_ERROR_UNEXPECTED_END when the JSON is invalid, because we reached the end of the
+ * JSON document too early
+ *         - #AZ_ERROR_UNEXPECTED_CHAR when the JSON is invalid, because of an invalid character
```

Comment:
Does this mean you validate the json_text span instead of trusting it? If so, the customer might not want to take this perf hit. Or, maybe it's a precondition?

Reply:
Since this is runtime behavior, it can't be a precondition. I ran this past @hihigupt and being safer by always validating the input is the better starting point, especially since payload sizes tend to be relatively small. If customers hit the perf issue, we can revisit this by either:
1) Adding a new overload with a "validation bool"
2) Remove the validation from this method (though I prefer not to do this).